### PR TITLE
[SG-42272] Migration from `shema.ts` type usage to `graphql-operations - client/web/src/enterprise/<batches, dotcom>`

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
Migrate typing using from `schema.ts` to `graphql-oparations`

## Success criteria

1. All imports from `schema.ts` in, the `client/web/entreprise<batches, dotcom>` folders,  are replaced with respective imports from `graphql-operations`.

## Ref
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/42272)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-42272)

## Test Plan
- Verify that all usage of `schema.ts` types has been migrated to their equivalent typing from `graphql-oparations.ts` file in `client/web` package for the folders: `batches, dotcom`

## App preview:

- [Web](https://sg-web-contractors-sg-42272.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-kvakggqyne.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/dotcom/productSubscriptions/ProductSubscriptionNode.tsx [hunk 1]
```diff
@@ -32,6 +32,10 @@ export const productSubscriptionFragment = gql`
         isArchived
         url
     }
+    fragment UserEmailFields on UserEmail {
+        email
+        verified
+    }
```
Comment (valerybugakov) [inline client/web/src/enterprise/dotcom/productSubscriptions/ProductSubscriptionNode.tsx:38]: Is the extraction of this fragment required? If not, let's revert this change.

**Review by valerybugakov — APPROVED** [review]

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.00% (-0.00 kb) | 0.00% (+0.38 kb) | 0.00% (+0.39 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/contractors/SG-42272/compare-report.html;tab=live_object) for a full comparison between the commits cc02e8a and 7b7e3a9 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by valerybugakov — APPROVED** [review]
Let's fix the Typescript check and merge this one.

Screenshot 2022-10-14 at 18 59 02
